src/components/BlueskyAuthProvider.tsx

```diff
@@ -76,23 +73,65 @@ export const BskyAuthProvider = ({ children }: Props) => {
     };
 
     initBsky();
-  }, []);
+  }, [bskyAuthClient]);
+
+  // for the custom resolver... another effect that runs when we use a PDS
+  useEffect(() => {
+    const client = new BrowserOAuthClient({
+      clientMetadata: blueskyClientMetadata,
+      handleResolver: serviceResolver,
+    })
+    setBskyAuthClient(client)
+  }, [serviceResolver])
+
+  const login = async (handle: string) => {
+    if (!bskyAuthClient) return;
+    try {
+      await bskyAuthClient.signIn(handle, {
+        scope: REQUIRED_ATPROTO_SCOPE
+      });
+    } catch (error) {
+      console.error("Login error:", error);
+    }
+  };
+
+  const logout = async () => {
+    if (!bskyAuthClient) return;
+    try {
+      bskyAuthClient.dispose()
+      setAuthenticated(false);
+      setSession(undefined);
+      setState(null);
```

Comment:
once you set userProfile I think you'll want to null that out too

Reply:
so, instead of `setSession(undefined)`, i should make it `setSession(null)`?